.evergreen/benchmarks.yml

```diff
@@ -0,0 +1,457 @@
+          ${PREPARE_SHELL}
+          for i in $(find ${DRIVERS_TOOLS}/.evergreen ${PROJECT_DIRECTORY}/.evergreen -name \*.sh); do
+            chmod +x $i
+          done
+
+  "init test-results":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          echo '{"results": [{ "status": "FAIL", "test_file": "Build", "log_raw": "No test-results.json found was created"  } ]}' > ${PROJECT_DIRECTORY}/test-results.json
+
+pre:
+  - func: "fetch source"
+  - func: "prepare resources"
+  - func: "windows fix"
+  - func: "fix absolute paths"
+  - func: "init test-results"
+  - func: "make files executable"
+  - func: "install dependencies"
+  - func: "download benchmark data"
+
+post:
+  - func: "stop mongo orchestration"
+  - func: "cleanup"
+
+tasks:
+  - name: "benchmark-4.2-standalone"
+    tags: ["4.2", "standalone"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "4.2"
+          TOPOLOGY: "server"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-4.2-replica_set"
+    tags: ["4.2", "replica_set"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "4.2"
+          TOPOLOGY: "replica_set"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-4.2-sharded_cluster"
+    tags: ["4.2", "sharded_cluster"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "4.2"
+          TOPOLOGY: "sharded_cluster"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-standalone"
+    tags: ["5.0", "standalone"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "server"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-replica_set"
+    tags: ["5.0", "replica_set"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "replica_set"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-sharded_cluster"
+    tags: ["5.0", "sharded_cluster"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "sharded_cluster"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-bson"
+    tags: ["bson"]
+    commands:
+      - func: "run bson benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-compile"
+    commands:
+      - func: "compile only"
+      - func: "upload benchmark results"
+
+axes:
+  - id: "mongodb-version"
+    display_name: MongoDB Version
+    values:
+      - id: "5.0"
```

Comment:
I opted for 5.0 and 4.2 instead of 5.0 and 3.6 as the design recommended to avoid having to deal with the involved database dropping mechanism required on sharded clusters. see: https://docs.mongodb.com/manual/reference/method/db.dropDatabase/#replica-set-and-sharded-clusters for more info